app/api/ada/lib/storage/database/genericBip44/tables.js

```diff
@@ -0,0 +1,391 @@
+// @flow
+
+import {
+  ConceptualWalletSchema,
+  KeySchema,
+} from '../uncategorized/tables';
+import { Type } from 'lovefield';
+import type { lf$schema$Builder } from 'lovefield';
+
+export type Bip44DerivationInsert = {|
+  PublicKeyId: number | null,
+  PrivateKeyId: number | null,
+  IsHardened: boolean,
+|};
+export type Bip44DerivationRow = {|
+  Bip44DerivationId: number, // serial
+  ...Bip44DerivationInsert,
+|};
+export const Bip44DerivationSchema: {
+  name: string,
+  properties: $ObjMapi<Bip44DerivationRow, ToSchemaProp>
+} = {
+  name: 'Bip44Derivation',
+  properties: {
+    Bip44DerivationId: 'Bip44DerivationId',
+    IsHardened: 'IsHardened',
+    PrivateKeyId: 'PrivateKeyId',
+    PublicKeyId: 'PublicKeyId',
+  }
+};
+
+export type Bip44WrapperInsert = {|
+  ConceptualWalletId: number,
+  IsBundled: boolean,
+  SignerLevel: number | null,
+  PublicDeriverLevel: number,
+  PrivateDeriver: number | null,
+  PrivateDeriverLevel: number | null,
+  Version: number,
+|};
+export type Bip44WrapperRow = {|
+  Bip44WrapperId: number, // serial
+  ...Bip44WrapperInsert,
+|};
+export const Bip44WrapperSchema: {
+  name: string,
+  properties: $ObjMapi<Bip44WrapperRow, ToSchemaProp>
+} = {
+  name: 'Bip44Wrapper',
+  properties: {
+    Bip44WrapperId: 'Bip44WrapperId',
+    ConceptualWalletId: 'ConceptualWalletId',
+    IsBundled: 'IsBundled',
+    SignerLevel: 'SignerLevel',
+    PublicDeriverLevel: 'PublicDeriverLevel',
+    PrivateDeriver: 'PrivateDeriver',
+    PrivateDeriverLevel: 'PrivateDeriverLevel',
+    Version: 'Version',
+  }
+};
+
+export type Bip44DerivationMappingInsert = {|
+  Parent: number,
+  Child: number,
+  Index: number,
+|};
+export type Bip44DerivationMappingRow = {|
+  Bip44DerivationMappingId: number, // serial
+  ...Bip44DerivationMappingInsert,
+|};
+export const Bip44DerivationMappingSchema: {
+  name: string,
+  properties: $ObjMapi<Bip44DerivationMappingRow, ToSchemaProp>
+} = {
+  name: 'Bip44DerivationMapping',
+  properties: {
+    Bip44DerivationMappingId: 'Bip44DerivationMappingId',
+    Parent: 'Parent',
+    Child: 'Child',
+    Index: 'Index',
+  }
+};
+
+export type PublicDeriverInsert = {|
+  Bip44WrapperId: number,
+  Bip44DerivationId: number,
+  Index: number,
+  Name: string,
+  LastBlockSync: number,
+|};
+export type PublicDeriverRow = {|
+  PublicDeriverId: number, // serial
+  ...PublicDeriverInsert,
+|};
+export const PublicDeriverSchema: {
+  name: string,
+  properties: $ObjMapi<PublicDeriverRow, ToSchemaProp>
+} = {
+  name: 'PublicDeriver',
+  properties: {
+    PublicDeriverId: 'PublicDeriverId',
+    Bip44WrapperId: 'Bip44WrapperId',
+    Bip44DerivationId: 'Bip44DerivationId',
+    Index: 'Index',
+    Name: 'Name',
+    LastBlockSync: 'LastBlockSync',
+  }
+};
+
+export type Bip44RootInsert = {|
```

Comment:
What is the purpose of this and all of the other **LEVEL** tables without attributes? Couldn't infer it from the specs.

Reply:
1) Easily be able to query for all of a certain level
2) In case we ever want to add an attribute later